### PR TITLE
Moves to finagle.httpx

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This is part of the larger project to upgrade Finagle
to Netty 4, which means getting rid of finagle-http
and moving everybody to finagle-httpx.

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
@taylorleese fyi

File: zipkin-example/src/main/scala/com/twitter/zipkin/example/Main.scala [hunk 1]
```diff
@@ -1,20 +1,18 @@
...
-import com.twitter.finagle.Http
-import com.twitter.finagle.stats.StatsReceiver
+import com.twitter.finagle.Httpx
 import com.twitter.server.{Closer, TwitterServer}
-import com.twitter.util.{Await, Closable, Future}
+import com.twitter.util.{Await, Closable}
 import com.twitter.zipkin.anormdb.AnormDBSpanStoreFactory
-import com.twitter.zipkin.collector.SpanReceiver
 import com.twitter.zipkin.common.Span
-import com.twitter.zipkin.{thriftscala => thrift}
```
Comment (codefromthecrypt) [inline zipkin-example/src/main/scala/com/twitter/zipkin/example/Main.scala:11]: sorry about the import change.. the amount of unused ones was driving me nuts. I don't feel like making a pull request just to do that.

File: zipkin-web/src/main/scala/com/twitter/zipkin/web/Handlers.scala [hunk 2]
```diff
@@ -55,7 +53,7 @@ class Handlers(jsonGenerator: ZipkinJson, mustacheGenerator: ZipkinMustache, que
 
     def apply(response: Response) {
       response.setContentType(typ)
-      response.content = ChannelBuffers.wrappedBuffer(content)
+      response.content = Buf.ByteArray(content)
```
Comment (codefromthecrypt) [inline zipkin-web/src/main/scala/com/twitter/zipkin/web/Handlers.scala:55]: Somewhere else in this file, this `content` is basically the result of an InputStream read into a byte array. Is there a shortcut to create a shareable Buf from an InputStream directly?

File: zipkin-web/src/main/scala/com/twitter/zipkin/web/Handlers.scala [hunk 3]
```diff
@@ -15,7 +14,6 @@ import com.twitter.zipkin.query.{SpanTimestamp, TraceCombo, TraceSummary, QueryR
 import com.twitter.zipkin.thriftscala.{Adjust, ZipkinQuery}
 import java.io.{File, FileInputStream, InputStream}
 import org.apache.commons.io.IOUtils
-import org.jboss.netty.buffer.ChannelBuffers
 import org.jboss.netty.handler.codec.http.{HttpRequest, HttpResponse}
```
Comment (taylorleese) [inline zipkin-web/src/main/scala/com/twitter/zipkin/web/Handlers.scala]: @adriancole remove ^^

File: zipkin-web/src/main/scala/com/twitter/zipkin/web/Handlers.scala (hunk 2 above)
Comment (taylorleese) [inline zipkin-web/src/main/scala/com/twitter/zipkin/web/Handlers.scala:55]: @dschobel @luciferous Can you help with Adrian's question?

Comment (luciferous) [inline zipkin-web/src/main/scala/com/twitter/zipkin/web/Handlers.scala:55]: Sorry if I'm narrow on the context. I'll look closer and follow up tomorrow, but here's a quick answer for now.

To get a Buf from an InputStream we have to use a Reader.

```
import com.twitter.io.Reader
val reader: Reader = Reader.fromStream(stream)
val buf: Future[Buf] = Reader.readAll(reader)
```

This gives a Future[Buf], which you could flatMap to get at the Buf.

**Comment by eirslett** [comment]
+1

File: zipkin-web/src/main/scala/com/twitter/zipkin/web/Handlers.scala (hunk 2 above)
Comment (codefromthecrypt) [inline zipkin-web/src/main/scala/com/twitter/zipkin/web/Handlers.scala:55]: have to admit that this is more complex than the existing code.. I think I'll leave it.

File: zipkin-web/src/main/scala/com/twitter/zipkin/web/Handlers.scala (hunk 3 above)
Comment (codefromthecrypt) [inline zipkin-web/src/main/scala/com/twitter/zipkin/web/Handlers.scala]: done. thanks!

**Pull request merged**